src/Neo/SmartContract/ContractState.cs

```diff
@@ -83,7 +83,7 @@ void IInteroperable.FromStackItem(StackItem stackItem)
             Id = (int)array[0].GetInteger();
             UpdateCounter = (ushort)array[1].GetInteger();
             Hash = new UInt160(array[2].GetSpan());
-            Nef = ((ByteString)array[3]).Memory.AsSerializable<NefFile>();
+            Nef = NefFile.Parse(((ByteString)array[3]).Memory, false);
```

Comment:
`ContractState` itself is pretty `public`, I fear it might be reused in some other place where the input would come from untrusted source. Likely right now we don't have anything like that, but if we could make it more specific to `ContractManagement` it'd be better.

Reply:
`RpcServer` and `StateService use it too, btw (but taking the data from storage).

Reply:
I can't think of any case where a contractState has to distrust the nefFile

Reply:
> where a contractState has to distrust the nefFile

Some *very* specific case: development environment where management storage may be setup manually with some specific contracts, like https://github.com/nspcc-dev/neo-go/issues/2406. So this technically is possible, but again, it's a very specific case.

Reply:
> But they download the NefFile from a trusted zone.

You're right, but it takes a single step from https://github.com/nspcc-dev/neo-go/issues/2406 to allow user to manually fill contract storage items with some garbage. But it's a nitpicking.

Reply:
Will be solved with the last changes